cdap-examples/SpamClassifier/src/main/java/co/cask/cdap/examples/sparkstreaming/SpamClassifier.java

```diff
@@ -0,0 +1,112 @@
+import co.cask.cdap.api.service.Service;
+import co.cask.cdap.api.service.http.AbstractHttpServiceHandler;
+import co.cask.cdap.api.service.http.HttpServiceRequest;
+import co.cask.cdap.api.service.http.HttpServiceResponder;
+import co.cask.cdap.api.spark.AbstractSpark;
+import com.google.common.base.Charsets;
+
+import java.net.HttpURLConnection;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Spark Streaming Example for classifying kafka messages through Spark MlLib NaiveBayesModel
+ */
+public class SpamClassifier extends AbstractApplication {
+
+  static final String SERVICE_HANDLER = "MessageClassification";
+  public static final String STREAM = "trainingDataStream";
+  public static final String DATASET = "messageClassificationStore";
+
+  @Override
+  public void configure() {
+    setName("SpamClassifier");
+    setDescription("A Spark Streaming Example for Kafka Message Classification");
+    addStream(new Stream(STREAM));
+    addSpark(new SparkSpamFilteringSpecification());
+    addService(SERVICE_HANDLER, new SpamClassifierServiceHandler());
+
+    // Store for message classification status
+    try {
+      ObjectStores.createObjectStore(getConfigurer(), DATASET, Double.class,
+                                     DatasetProperties.builder().setDescription("Kafka Message Spam " +
+                                                                                  "Classification").build());
+    } catch (UnsupportedTypeException e) {
+      // This exception is thrown by ObjectStore if its parameter type cannot be
+      // (de)serialized (for example, if it is an interface and not a class, then there is
+      // no auto-magic way deserialize an object.) In this case that will not happen
+      // because Double is actual classes.
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * A Spark Streaming program which classifies kafka messages as spam or ham
+   */
+  public static final class SparkSpamFilteringSpecification extends AbstractSpark {
+    @Override
+    public void configure() {
+      setName("SpamClassifierProgram");
+      setDescription("Spark Streaming Based Kafka Message Classifier");
+      setMainClass(SpamClassifierProgram.class);
+      setDriverResources(new Resources(1024));
+      setExecutorResources(new Resources(1024));
+    }
+  }
+
+  /**
+   * A {@link Service} handler to get the classification of kafka messages
+   */
+  public static final class SpamClassifierServiceHandler extends AbstractHttpServiceHandler {
+
+    static final String CLASSIFICATION_PATH = "classification";
+    static final String SPAM = "Spam";
+    static final String HAM = "Ham";
+
+    @UseDataSet(DATASET)
+    private ObjectStore<Double> messageClassificationStore;
+
+    @Path(CLASSIFICATION_PATH + "/{message-id}")
+    @GET
+    public void centers(HttpServiceRequest request, HttpServiceResponder responder, @PathParam("message-id") String
+      messageId) {
```

Comment:
This wrapping is awkward. Better align the `@PathParam` with the `HttpServiceRequest`.